### PR TITLE
Update confirmation modal after progressive onboarding

### LINKED ISSUE #9873: [WC Payments NOX] Change onboarding success modals
### Description
Currently progressive onboarding success modal looks like this:
Image

We should modify it according to the design:
Image

### Designs
Design post: pfHfG4-q7-p2
Figma: xO8YdpKfaVoP2d8NvweZ5I-fi-1428_21437

### Dev notes
Reactifying the Payments settings page: pe4Hk0-1ly-p2
Populate react-powered main payments settings screen PR: https://github.com/woocommerce/woocommerce/pull/50825

### Additional context
PRD: 1wMfX8CAjDnOp24tcU-PADB0pOt3cz7mGdyZYeRBxx2Y-gdoc

### PR DESCRIPTION
**Pull request by elazzabi**
Fixes #9873

#### Changes proposed in this Pull Request

This changes the confirmation modal after the onboarding according to the designs in #9873 

#### Testing instructions

* Make [this variable](https://github.com/Automattic/woocommerce-payments/blob/develop/client/overview/index.js#L112) `true`
* Remove [this line](https://github.com/Automattic/woocommerce-payments/blob/develop/client/overview/modal/progressive-onboarding-eligibility/index.tsx#L71)
* Visit `Payments > Overview`, you should see a modal:

Screenshot 2024-12-10 at 16 05 38

Screenshot 2024-12-10 at 16 05 31

Please check both mobile/desktop versions.

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9915` or branch name `add/update-confirmation-modal-nox` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/update-confirmation-modal-nox) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: da3464ea1ba3c2ce06457b20b86fa2cacdf52712
- Build time: 2024-12-10 17:01:30 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/overview/modal/progressive-onboarding-eligibility/index.tsx [hunk 1]
```diff
@@ -59,75 +60,74 @@ const ProgressiveOnboardingEligibilityModal: React.FC = () => {
...
+							'Provide some additional details about your business so you can continue accepting payments and begin receiving payouts without restrictions.',
+							'woocommerce-payments'
+						) }
+					</div>
 				</div>
 			</div>
 			<div className="wcpay-progressive-onboarding-eligibility-modal__footer">
 				<Button variant="secondary" onClick={ handleSetup }>
-					{ __( 'Start receiving payouts', 'woocommerce-payments' ) }
+					{ __( 'Set up deposits', 'woocommerce-payments' ) }
```
Comment (dmallory42) [inline client/overview/modal/progressive-onboarding-eligibility/index.tsx]: WooPayments now uses `payouts` everywhere where `deposits` would normally be used. Maybe the designs were done before this change was made, but we should keep this as `payouts` for consistency. (So, `Set up payouts`).

File: client/overview/modal/progressive-onboarding-eligibility/test/index.test.tsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ describe( 'Progressive Onboarding Eligibility Modal', () => {
 
 		user.click(
 			screen.getByRole( 'button', {
-				name: 'Start receiving payouts',
+				name: 'Set up deposits',
```
Comment (dmallory42) [inline client/overview/modal/progressive-onboarding-eligibility/test/index.test.tsx]: Same comment here about using `payouts` over `deposits`.

**Review by dmallory42 — APPROVED** [review]
Looks good to me, thanks for working on it! I left one comment I think we should address, and then this is good to go! :shipit:

File: client/overview/modal/progressive-onboarding-eligibility/index.tsx (hunk 1 above)
Comment (elazzabi) [reply client/overview/modal/progressive-onboarding-eligibility/index.tsx]: Oh yes, 100%!

File: client/overview/modal/progressive-onboarding-eligibility/index.tsx [hunk 3]
```diff
@@ -59,75 +60,74 @@ const ProgressiveOnboardingEligibilityModal: React.FC = () => {
 		setModalVisible( false );
 	};
 
-	// Workaround to remove Modal header from the modal until `hideHeader` prop can be used.
-	useEffect( () => {
```
Comment (mordeth) [inline client/overview/modal/progressive-onboarding-eligibility/index.tsx:63]: Removing the useEffect triggered linting error `'useEffect' is defined but never used`.

**Review by mordeth — APPROVED** [review]
The code changes look great. I’ve noted a minor nitpick.

**Pull request merged**